p2p/net/swarm/dial_worker.go

```diff
@@ -130,7 +140,8 @@ func (w *dialWorker) loop() {
 		}
 		timerRunning = false
 		if dq.Len() > 0 {
-			if dialsInFlight == 0 && !w.connected {
+			top := dq.top()
+			if !top.ForceDelay && dialsInFlight == 0 && !w.connected {
```

Comment:
Instead of this, how about we introduce another queue:
fixedDelayQueue. 

1. This construct forces us to expose ForceRetry, which external users never need to use. 
2. Makes the semantics slightly incorrect for dials that are behind in the queue but don't have the forceRetryFlag. 

I suggest introducing:
- a separate fixed Delay Queue with its separate fixedDelayTimer
- adding a `now time.Time` argument to NextBatch that'll return elements that have expired by now
- Have two blocks for handling dials, 1 for dialQueue and 1 for fixedDelayDialQueue. 

I don't feel strongly about the suggestion. Anything that fixes 2 would be great.